assets/js/SignPanel.tsx

```diff
@@ -161,58 +161,50 @@ interface SignPanelProps {
   readOnly: boolean;
 }
 
-class SignPanel extends React.Component<
-  SignPanelProps,
-  {
-    staticLine1: string;
-    staticLine2: string;
-    customChanges: boolean;
-    initialTime: number;
-    confirmingUngroup: boolean;
-  }
-> {
-  constructor(props: SignPanelProps) {
-    super(props);
-
-    this.saveStaticText = this.saveStaticText.bind(this);
-    this.handleInputLine1 = this.handleInputLine1.bind(this);
-    this.handleInputLine2 = this.handleInputLine2.bind(this);
-
-    this.state = {
-      staticLine1: props.signConfig.line1 || '',
-      staticLine2: props.signConfig.line2 || '',
-      customChanges: false,
-      initialTime: props.currentTime,
-      confirmingUngroup: false,
-    };
-  }
-
-  handleInputLine1(newText: string): void {
-    this.setState({ staticLine1: newText, customChanges: true });
+function SignPanel({
+  alerts,
+  signId,
+  modes = {
+    auto: true,
+    custom: true,
+    headway: true,
+    off: true,
+  },
+  signContent,
+  currentTime,
+  line,
+  setConfigs,
+  signConfig,
+  realtimeId,
+  signGroup,
+  ungroupSign,
+  readOnly,
+}: SignPanelProps): JSX.Element {
+  const [staticLine1, setStaticLine1] = React.useState(signConfig.line1 || '');
+  const [staticLine2, setStaticLine2] = React.useState(signConfig.line2 || '');
+  const [customChanges, setCustomChanges] = React.useState(false);
```

Comment:
This might be better named `hasCustomChanges`, else it sort of sounds like it _is_ the custom changes.